webapp/src/components/NavBar.tsx

```diff
@@ -1,32 +1,138 @@
-import React from 'react';
-import Navbar from 'react-bootstrap/Navbar';
-import Nav from 'react-bootstrap/Nav';
-import Container from 'react-bootstrap/Container';
+import * as React from 'react';
+
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import AccountCircle from '@mui/icons-material/AccountCircle';
+import IconButton from '@mui/material/IconButton';
+import MenuIcon from '@mui/icons-material/Menu';
+import HomeIcon from '@mui/icons-material/Menu';
+import Badge from '@mui/material/Badge';
+import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
+import SwipeableDrawer from '@mui/material/SwipeableDrawer';
+import MenuList from '@mui/material/MenuList';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+
 import {Link} from 'react-router-dom';
-import Shopping from './Shopping';
-import LinkContainer from 'react-router-dom';
-
-function NavBar(): JSX.Element{
-    return (
-        <>
-            <br />
-            <Navbar bg="dark" variant="dark">
-                <Container>
-                    <Navbar.Brand>Navbar</Navbar.Brand>
-                    <Nav className="me-auto">
-                    <Nav.Link >
-                        <Link to="/">Home</Link>
-                    </Nav.Link>
-                        
-                    <Nav.Link>
-                        <Link to="/cart">Cart
-                        </Link>
-                    </Nav.Link>
-                    </Nav>
-                </Container>
-            </Navbar>
-        </>
-    );
+
+function NavBar(props : any): JSX.Element {
+    const [auth, setAuth] = React.useState(true);
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+    const [state, setState] = React.useState(false);
+
+    const handleMenu = (event: React.MouseEvent<HTMLElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const toggleDrawer = 
+        (open: boolean) =>
+        (event: React.KeyboardEvent | React.MouseEvent) => {
+            if (event &&
+                event.type === 'keydown' &&
+                ((event as React.KeyboardEvent).key === 'Tab' ||
+                (event as React.KeyboardEvent).key === 'Shift')
+            )
+                return;
+        
+            setState(open);
+        };
+    
+      const list = () => (
+        <Box
+            onClick={ toggleDrawer(false) } 
+            onKeyDown={ toggleDrawer(false) }
+        >
+            <MenuList>
+                <MenuItem onClick={event =>  window.location.href='/'} >
```

Comment:
You should add here the link component like in the cart item, because if you go to the cart, click home, and the go back to the cart all the products disappear, and I think is better to prevent this